V-Slice Sound Tray

hackx2 opened the pull request:
# Changes
- V-Slice Sound Tray
- Sound Tray Supports Mod Folders (sounds, images)
- Only works when the BASE_GAME_FILES allowed thingy is enabled.

# Showcase
![image](https://github.com/user-attachments/assets/bf03649d-7143-48a6-bea8-1fe82e5c86fc)

hackx2 committed:
V-Slice Sound Tray

+ Change Funkin.vis version in setup files
+ V-Slice Sound Tray

hackx2 committed:
Fixes

wlshedd commented:
uhh i tried this on 0.7.3 and i got this error

source/backend/FunkinSoundTray.hx:133: characters 12-89 : Not enough arguments, expected key:String, library:Null<String>
source/backend/FunkinSoundTray.hx:141: characters 13-66 : Not enough arguments, expected key:String, library:Null<String>

Hackx2 commented:
Send me your `returnSound` function(Paths.hx) please :3

wlshedd commented:
i think this is it.
![Desktop Screenshot 2024 12 05 - 13 00 24 46](https://github.com/user-attachments/assets/b946f5aa-8df7-4d08-af9e-00201b722b21)

hackx2 committed:
uhh fix

Hackx2 commented:
Lemme have a look!

wlshedd commented:
ok thanks

hackx2 committed:
Remove Thingy

hackx2 committed:
fix

only works with base_game_files enabled.
Hopefully fixed legacy psych `Paths.returnSound`
idk lol

hackx2 committed:
Update FunkinSoundTray.hx

Hackx2 commented:
> ok thanks

Uhh hey! 👋👋👋

Didn't test, but hopefully this works !!!

I've added a function that checks PsychEngine's version, and if it's below 1.0, it uses the `returnSound` function that's in your version of Psych!

Sorry... I'm awful at explaining.

I hope this helps. :3

hackx2 committed:
Update ChartingState.hx

hackx2 committed:
revert some stuff

hackx2 committed:
revert more stuff lol

hackx2 committed:
bruh

wlshedd commented:
uhh it compiles now however theres no change to the volume slider, its just the old one.

wlshedd commented:
ok i see, do i replace all 36 key:string with it or just the one?

wlshedd commented:
ok nvm i forgot about BASE_GAME_FILES lol thanks for the help!

Hackx2 commented:
np
